packages/connection-manager/src/js/ConnectionStore/ConnectionStore.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Connection Queue which holds queued connections.
+ *
+ * @todo check if optimisation of storage is needed (performance tests needed)
+ */
+export default class ConnectionStore {
+  /**
+   * Initialises an instance of ConnectionStore
+   * @constructor
+   */
+  constructor() {
+    var store = [];
+    Object.defineProperty(this, "store", {
+      get: () => store
+    });
+  }
+  /**
+   * Adds given connection with given (or default) priority to queue
+   * @param {AbstractConnection} connection – Connection to be added to queue
+   * @param {Integer} priority – priority of connection
+   */
+  add(connection, priority) {
```

Comment:
We don't, its removed from connection because they dont need to know what priority they have.

if we add it to connection, we also need to add some kind of listener if the priority of a connection changes, because then we need to reorder the store. but we could do that if we are adding events to the store anyways..